src/transformers/models/owlvit/modeling_owlvit.py

```diff
@@ -937,23 +1070,13 @@ def get_text_features(
         >>> text_features = model.get_text_features(**inputs)
         ```"""
         # Use OWL-ViT model's config for some fields (if specified) instead of those of vision & text components.
-        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
-        output_hidden_states = (
-            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
-        )
```

Comment:
Why are we removing standard functionality in this model?

Reply:
This was actually a mistake in the original CLIP implementation - the public API of this method includes "output_attentions" and "output_hidden_states", but even if you set them to True, the method only returns the text features. Hence we should actually remove "output_attentions", "output_hidden_states" and "return_dict" from the public API here, as they don't have any use.

Of course we should handle this with deprecation warnings.

This could be done for all CLIP variants, including GroupViT, X-CLIP.

cc @patil-suraj